### ISSUE TITLE
syntax-highlighting in 'source' block

### ISSUE THREAD

**Issue by byaka** [Moeditor/Moeditor#31]
![image](https://cloud.githubusercontent.com/assets/1580811/17066126/44c8e266-504e-11e6-9398-dac2c66626b5.png)

also u can see incorrect scroll-sync in this screenshot

**Comment by lucaschimweg** [comment]
Heading not really related to issue...

**Comment by byaka** [comment]
why?
in block "source (python)" syntax-highlighting working incorrect, and python's comments highlighted as headers.

**Comment by Menci** [comment]
What's the expected result? Did you mean this

    ```code```
    # title

Should be parse to a inline code?

But GitHub parse this to a code block and 'code' as the info string

```code```
# title

**Comment by Menci** [comment]
According to Commonmark spec, 

    ``` code ```

Should be parse to a inline code.

Maybe there's something wrong with CodeMirror.

**Comment by Menci** [comment]
And you mentioned the wrong sync scroll. I tried to fix it in a recent commit https://github.com/Moeditor/Moeditor/commit/d98bf0695bf84f5e8dc7e7404d432ced283429f1. I will make a new release soon.

**Comment by byaka** [comment]
> What's the expected result? Did you mean this

> ...

> Should be parse to a inline code.

No, in my screenshot i have `# comment` in code-block.

and `# comment` highlighted as header, but it part of multiline code block.

**Comment by tmr232** [comment]
Would also be nice to have actual syntax highlighting in code blocks.

**Comment by Menci** [comment]
I think you mean

    ```pip install XXX```

should be a inline code. And

    ```python

should starts a code block.

**Comment by byaka** [comment]
i think we say about different things. I say about highlight in editor (not results).
On screenshot first visible header `###install` is true highlighted, but other "headers" is not really headers - they are comments in multiline code-block. But they also highlighted as headers.

In results all rendered correctly (but it not visible in screenshot due scroll problem).

**Comment by byaka** [comment]
ok, i find problem. if use one-quoted syntax for inline-code, highlighting works properly. but 3-quoted syntax break editor (but result also work properly).

see first lines in files

[correct.txt](https://github.com/Moeditor/Moeditor/files/388442/correct.txt)
[wrong.txt](https://github.com/Moeditor/Moeditor/files/388443/wrong.txt)

**Comment by byaka** [comment]
but github (and markdown also) allow to use 3-quoted syntax for inline-code.

**Comment by tmr232** [comment]
Seems like an odd parsing issue. Adding a character before the inline code line seems to fix it.

**Comment by Menci** [comment]
Maybe a bug in CodeMirror https://github.com/codemirror/CodeMirror/issues/4137.

**Comment by tmr232** [comment]
@Menci I don't think it is, as it works OK here  - https://codemirror.net/mode/markdown/

**Comment by Menci** [comment]
The *mode* we're using is NOT markdown, is GFM.

**Comment by tmr232** [comment]
Okay, checked again - definitely codemirror is to blame. https://codemirror.net/mode/gfm/index.html

**Comment by Menci** [comment]
See this issue https://github.com/codemirror/CodeMirror/issues/4137.

My test code is:

    ``` aa ```
    foo
    # title
